[Timothy Soh] iP

LargeCrowd opened the pull request:
# CowboyBebot (aka Bebot)
> "Its just a chatbot"

Bebot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [NOT here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X]  Managing tasks
- [X]  Managing deadlines
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

LargeCrowd committed:
Update Duke.java

Level-0 changes. Program exits after greeting with new name

LargeCrowd committed:
Update Duke.java

Level-1 changes. Takes an input then repeats it

LargeCrowd committed:
Level 2: Add, List

LargeCrowd committed:
level 3.1

LargeCrowd committed:
level-3. Mark as Done

LargeCrowd committed:
changed "item" to "task"

LargeCrowd committed:
Starting level 4

LargeCrowd committed:
debug level-3

LargeCrowd committed:
added documentation and other small bug fixes

LargeCrowd committed:
Level 4

LargeCrowd committed:
A-TextUiTesting

LargeCrowd committed:
Level 5

LargeCrowd committed:
Level 6: Delete, and did better organisation of task adding

LargeCrowd committed:
A-Enums

LargeCrowd committed:
Edit documentation & styling

LargeCrowd committed:
Start level 7

LargeCrowd committed:
Reorganise and restart level 7

LargeCrowd committed:
Create tasklist.txt

LargeCrowd committed:
Add ability to save list, but doesnt load data.

LargeCrowd committed:
Need to fix tasklist display.

LargeCrowd committed:
Work on loading tasks.

LargeCrowd committed:
Test branch-level-7 pushing

LargeCrowd committed:
Test branch-level-7 pushing

LargeCrowd committed:
Complete level 7

LargeCrowd committed:
Small changes.

LargeCrowd committed:
Merge branch 'branch-Level-7'

LargeCrowd committed:
Small if logic change

LargeCrowd committed:
Merge branch 'branch-Level-7'

LargeCrowd committed:
Better encapsulation and abstraction.

LargeCrowd committed:
Reset level 8

LargeCrowd committed:
Reset level 8-2

LargeCrowd committed:
Add date formatting to Deadline

Not yet added to Events. Deadline accepts 4 formats of dates and displays as "d MMM yyyy"

LargeCrowd committed:
Move date parsing logic to Task class

LargeCrowd committed:
Add Ui class. Work on Storage Class

LargeCrowd committed:
Work on the new classes.

LargeCrowd committed:
Complete MoreOOP

LargeCrowd committed:
Divide classes into packages.

LargeCrowd committed:
Merge pull request #1 from LargeCrowd/A-Packages

Divide classes into packages.

LargeCrowd committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

LargeCrowd committed:
Debug Storage.

Incomplete

LargeCrowd committed:
Update storage logic.

dillontkh commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Override
+    protected String taskType() {
```
We should use verbs for method names in accordance to the [CS2103 Java Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html#naming). I noticed this issue elsewhere too.

```suggestion
    protected String getTaskType() {
```

dillontkh commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,109 @@
+public class Task {
+    protected String task;
+    protected boolean isDone;
+    protected DateTimeFormatter f1 = DateTimeFormatter.ofPattern("d/M/yyyy", Locale.ENGLISH);
+    protected DateTimeFormatter f2 = DateTimeFormatter.ofPattern("yyyy-M-d", Locale.ENGLISH);
+    protected DateTimeFormatter f3 = DateTimeFormatter.ofPattern("d MMM yyyy", Locale.ENGLISH);
+    protected DateTimeFormatter f4 = DateTimeFormatter.ofPattern("d-M-yyyy", Locale.ENGLISH);
```
Perhaps use clearer variable names? `f1` to `f4` doesn't convey much meaning 😅

LargeCrowd replied:
it felt neater to give ` f1` rather than `format1`, which is why i made this choice, do lmk if you still think its more appropriate to do `format1` though

dillontkh replied:
As someone who has recently done this increment I know it's because they asked us to accept and display the dates in different formats, but to someone who isn't familiar with the application requirements this might be a little confusing.

What I did was name them something like `UserInputFormat`, `DisplayFormat`, `SaveFormat` etc. This way, it provides a little bit more information on the intended usage of these formats rather than just `format1`, `format2`, etc.

LargeCrowd replied:
okay gets. actually i just put them all in as like kinda edge cases of other date formats being input. like giving it more options to not throw an exception but okay ill see how i can rename them a bit better

dillontkh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,41 @@
+        this.taskList = taskList;
+    }
+
+    public boolean parseInput(String input) {
```
Maybe the method name could be more descriptive? 

It appears that you are handling the detection and execution of the command here. Maybe something like `executeUserInput()`?

dillontkh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+                        break;
+                    default:
+                        System.out.println("Unrecognized task type: " + task);
+                        throw new IOException("Error loading data from file: unrecognized task type");
```
Should `case` statements be indented according to the [CS2103 Java Coding Standards](https://se-education.org/guides/conventions/java/intermediate.html#layout)?

dillontkh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,238 @@
+        } else if (task.startsWith("delete")) {
+            return TaskType.DELETE;
+        } else {
+            return TaskType.UNKNOWN;
```
Was there a reason you chose to explicitly include an `UNKNOWN` task type? I thought the `default` part of your `switch` statement would have already caught any invalid commands.

LargeCrowd replied:
youre right thats an oversight on my part

dillontkh reviewed (commented):
Overall, code was easy to understand. Variables and methods were given descriptive names apart from a few minor exceptions. You might want to consider using checkstyle to catch any CS2103 Java Coding Standard violations. Keep up the good work!

LargeCrowd committed:
Fix issues based on PR.

Renamed the taskType() methods to getTaskType() to follow the coding standard. Renamed parseInput() in Parser.java to executeUserInput() to show better description. Reformatted the switch/case block to follow standard

LargeCrowd committed:
Add JUnit for deadline and todo classes.

LargeCrowd committed:
Create MANIFEST.MF

LargeCrowd committed:
Add javadocs.

LargeCrowd committed:
Follow coding conventions.

LargeCrowd committed:
Merge branch 'branch-A-CodingStandard'

LargeCrowd committed:
Add ability to find by keyword.

Includes minor javadoc edits

LargeCrowd committed:
Add help line for Find

LargeCrowd committed:
Rename the chatbot from Duke to CowboyBebot

LargeCrowd committed:
Add checkstyle.xml and suppressions.xml

LargeCrowd committed:
Update build.gradle

LargeCrowd committed:
Fix CheckStyle errors

LargeCrowd committed:
Update settings.json

LargeCrowd committed:
Implement JavaFX tutorial part 3

LargeCrowd committed:
Start JavaFX part 4

LargeCrowd committed:
Add fxml files

DialogBox methods in CowboyBebot not working

LargeCrowd committed:
Errors with this implementation of Part 4

LargeCrowd committed:
Reset to part 3

LargeCrowd committed:
Revert "Implement JavaFX tutorial part 3"

This reverts commit 798c892fd59229b49e74ed46615e83ecad0c92f7.

LargeCrowd committed:
Reset before Level 10

LargeCrowd committed:
Level 10 part 1

LargeCrowd committed:
Add level 10 part 2

LargeCrowd committed:
Level 10 part 3

LargeCrowd committed:
Level 10 part 3

LargeCrowd committed:
Up to DialogBox in level 10 part 3

LargeCrowd committed:
Add level 4 incomplete

Renamed the chatbot to CatChat
Trying to get the DialogBox working with the original code. In CatChat.java, trying to get getUserInput working properly

LargeCrowd committed:
Continue level 20

Created TaskListGui to change TaskList class such that print methods return strings
Moved Delete and Help to CommandType instead of TaskType

LargeCrowd committed:
Complete level 10

Not sure how to create a greeting on program start up

LargeCrowd committed:
Clean up code variables

LargeCrowd committed:
Clean up code

LargeCrowd committed:
Create JAR file

LargeCrowd committed:
Update Ui.java

LargeCrowd committed:
SLAP based on problems in email

Removed dead code from Ui.java.
Improved SLAP for CatChat.java using
abstraction.

LargeCrowd committed:
Add minor code quality things

Tried to add greeting on startup failed.
Refactored duke to catChat in Main.
Added title in Main.

LargeCrowd committed:
Add greeting functionality

Successfully added the ability to greet when program starts

LargeCrowd committed:
Edit Scenebuilder

Edited the GUI window in scenebuilder.
Mde the helplist easier to read.
Added one assert to MainWindow.java.

LargeCrowd committed:
Merge branch 'branch-A-Assertions'

LargeCrowd committed:
Add getters for Storage

Added task description getters in Storage.

LargeCrowd committed:
Fix bug in deleting tasks

Fixed a bug where exception is thrown
despite successfully deleting tasks.

LargeCrowd committed:
Add C-DetectDuplicates extension

Removed description getters from
Storage.
Added extension C-DetectDuplicates

LargeCrowd committed:
Remove failed duplicate detector

Couldn't figure out how to do detect
duplicates so removed it.
Also, refactored some small things
for clarity.

LargeCrowd committed:
Attempt implement C-Help

LargeCrowd committed:
Update README.md

LargeCrowd committed:
Remove gethelp2

Returned gethelp2() to original
one getHelp() method

LargeCrowd committed:
Update README.md

Added list of commands

LargeCrowd committed:
Update README.md

LargeCrowd committed:
Update README.md

LargeCrowd committed:
Update README.md

LargeCrowd committed:
Merge branch 'master' of https://github.com/LargeCrowd/ip

LargeCrowd committed:
Create Ui.png

LargeCrowd committed:
Update Ui.png

LargeCrowd committed:
Create ParserTest.java

Created two testing methods for Parser using JUnit

LargeCrowd committed:
Correct JAR file parameters in build.gradle

LargeCrowd committed:
Rearrange load tasklist logic

LargeCrowd committed:
Correct README.md

User Guide was previously placed in
the wrong folder
